ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,189 @@
+      return mat.derived().reshaped().eval();
+    }
+  };
+
+  template <bool strainlike = false>
+  struct MatricizeWithVoigt
+  {
+    template <typename Derived, int RowsAtCompileTime, int ColsAtCompileTime>
+    static auto transform(const Eigen::DenseBase<Derived>& vec, int rows = RowsAtCompileTime,
+                          int cols = ColsAtCompileTime) {
+      assert(rows == RowsAtCompileTime && cols == ColsAtCompileTime &&
+             "Only the fixed size values work for voigt matrices and vectors");
+      static_assert(RowsAtCompileTime != Eigen::Dynamic and ColsAtCompileTime != Eigen::Dynamic,
+                    "Voigt notation only available for fixed size vectors and matrices");
+      return fromVoigt(vec.derived(), strainlike);
+    }
+  };
+
+  struct MatricizeGeneric
+  {
+    template <typename Derived, int RowsAtCompileTime, int ColsAtCompileTime>
+    static auto transform(const Eigen::DenseBase<Derived>& vec, int rows = RowsAtCompileTime,
+                          int cols = ColsAtCompileTime) {
+      return vec.derived().reshaped(Eigen::fix<RowsAtCompileTime>(rows), Eigen::fix<ColsAtCompileTime>(cols)).eval();
+    }
+  };
+
+} // namespace Impl
+
+namespace ResultType {
+#define REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, MaxRowsExpr, MaxColsExpr, VectorizeStruct, \
+                                 MatricizeStruct)                                                           \
+  template <typename ScalarType, int gridDim, int worldDim>                                                 \
+  struct structName                                                                                         \
+  {                                                                                                         \
+    friend auto toString(structName) { return #structName; }                                                \
+                                                                                                            \
+    using type       = Eigen::Matrix<ScalarType, rowsExpr, ColsExpr, 0, MaxRowsExpr, MaxColsExpr>;          \
+    using Vectorizer = VectorizeStruct;                                                                     \
+    using Matricizer = MatricizeStruct;                                                                     \
+  }
+
+#define REGISTER_SYMMETRIC_RESULTTYPE(structName, rowsExpr, ColsExpr, MaxRowsExpr, MaxColsExpr) \
+  REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, Ikarus::Util::VectorizeWithVoigt,    \
+                           Ikarus::Util::MatricizeWithVoigt)
+
+#define REGISTER_SIMPLE_SYMMETRIC_RESULTTYPE(structName, rowsExpr, ColsExpr, strainlike) \
+  REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, rowsExpr, ColsExpr,           \
+                           Ikarus::Impl::VectorizeWithVoigt<strainlike>, Ikarus::Impl::MatricizeWithVoigt<strainlike>)
+
+#define REGISTER_RESULTTYPE(structName, rowsExpr, ColsExpr, MaxRowsExpr, MaxColsExpr)      \
```

Comment:
I think this is more or less ready from my side. 
We should also add some bits of documentation to these Macros on how they can be used. 
Also is `simple` a good name here?